src/mission_companion.cpp

```diff
@@ -466,6 +575,63 @@ bool talk_function::outpost_missions( npc &p, std::string id, std::string title
     if (cur_key == "Recover Ally from Menial Labor"){
         labor_return(p);
     }
+
+    if (cur_key == "Upgrade Camp"){
+        const recipe *making = &recipe_id( bldg ).obj();
```

Comment:
Please use references and not pointers in situations like this. `obj()` already returns a reference.